src/main/java/com/mindee/model/fields/BaseField.java

```diff
@@ -10,20 +11,20 @@ public abstract class BaseField {
   private final Boolean reconstructed;
   private final String rawValue;
   private final Double confidence;
-  private final List<List<Double>> polygon;
-  private final List<List<Double>> bbox;
+  private final Polygon polygon;
+  private final Polygon bbox;
```

Comment:
This should ideally be of type BoundingBox

Reply:
@ianardee ask to not do that because Mindee do not use the usual concept of bouding box. 
But it use instead a polygon which have only four points.

I let him argue about that

Reply:
Indeed, the API always returns a list of points, it never exposes the 4-coordinate bbox type.

Also, our front end libraries (react and vue.js) only deal with polygon types (list of points).

As I see it, we have two options:

### 1. Follow BBOX standards

`private final Polygon bbox;` becomes `private final BoundingBox bbox;`

Add a method `asPolygon()` to the `BoundingBox` class, which converts from standard 4 coord system to 4 Point Polygon system.

We use this conversion when we need to interact with the Mindee FE tools.

Internally we should always use the Polygon class, so would need to convert to Polygon class to use most of our geometry functions.

pro: this is probably what people are expecting when they ask for a bounding box.
con: requires us to do back and forth conversions in some cases

### 2. Use Mindee standard
We always manipulate everything as Polygon class. We can provide a `getBbox()` method to the `Polygon` class for those that want this format.

pro: easier to code and to reason about IMHO
con: could be unexpected for the end user to get a Polygon when calling the bbox method.

### Food for thought
4-coordnate system is really the accepted standard everywhere, even Wikipedia [defines it as such](https://en.wikipedia.org/wiki/Minimum_bounding_rectangle):
> bounding box is an expression of the maximum extents of a two-dimensional object within its x-y coordinate system in other words: min(x), max(x), min(y), max(y)

### Conclusion
I initally wanted to go with option 2 for ease of coding and limiting changes we would need to do.

Now, not so sure... If option 2 is not a lot of extra work, could be better for the end user.

@Kvindee @Vivek-mindee @jonathanMindee any thoughts?

Reply:
### Terms:
BoundingBox => Mindee "standard", list of 4 points
BBox => Standard 4 coordinate system

### TODO:
As discussed, remove all notion of BBox, including the current `BoundingBox` class and associated utils.
(if we were to keep the `BoundingBox` class, it would need to be renamed to `BBox`)

Rename `bbox` property to `boundingBox` and use a `Polygon` class to represent the data.